lib/slack-ruby-bot/hooks/hook_support.rb

```diff
@@ -0,0 +1,41 @@
+module SlackRubyBot
+  module Hooks
+    module HookSupport
+      def self.included(base)
+        base.cattr_accessor :hook_blocks
```

Comment:
in this case, it's just a collection of lambdas to hold the `on 'event_name'` execution blocks, once the class gets instantiated this gets flushed to the `Hook::Set` . Not that I'm super happy about that because we explicitly need to call the flush method.

thoughts?